src/main/java/xyz/corman/velt/Utils.java

```diff
@@ -157,13 +156,7 @@ public static Value getBindings(Context context) {
 
 	public static <T> VoidRun wrap(Callable<T> val) {
 		VoidRun res = new VoidRun((Object ...args) -> {
-			new BukkitRunnable() {
-				@Override
-				public void run() {
-					val.execute(args);
-				}
-
-			}.runTaskLater(Velt.getInstance(), 0);
+			Bukkit.getScheduler().scheduleSyncDelayedTask(Velt.getInstance(), val::execute, 0); //why does this need to be delayed by 1 tick?
```

Comment:
It needs to be delayed by one tick because that's the entire point of the method? It schedules a task that runs on another thread back to the main one that JS runs on so Graal doesn't throw an error.

Reply:
Forgot to reply. But I was just confused why you would want to go with 1 tick, you could've done a 0 tick wait with the scheduler iirc.